hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMBlockProtocolServer.java

```diff
@@ -198,6 +201,13 @@ public List<AllocatedBlock> allocateBlock(
             .allocateBlock(size, replicationConfig, owner, excludeList);
```

Comment:
maybe can add a new subclass for `PipelineChoosePolicy` (such as `NearestPipelineChoosePolicy`) to "sort" the datanode when `choosePipeline`? This would get a better results.
 But this should be a new PR.

Reply:
Thank you for the idea. Agreed, we can have a locality aware pipeline choose policy to pick the nearest pipeline on top of sorting the datanodes.

Reply:
Let's do it separately since this PR is already not small.

Reply:
I have created https://issues.apache.org/jira/browse/HDDS-9867

Feel free to work on this if interested.